ext/pdo_mysql/mysql_statement.c

```diff
@@ -619,6 +619,17 @@ static int pdo_mysql_stmt_param_hook(pdo_stmt_t *stmt, struct pdo_bound_param_da
 static int pdo_mysql_stmt_fetch(pdo_stmt_t *stmt, enum pdo_fetch_orientation ori, zend_long offset) /* {{{ */
 {
 	pdo_mysql_stmt *S = (pdo_mysql_stmt*)stmt->driver_data;
+
+	if(S->stmt) {
+		MYSQL_RES     *prepare_meta_result;
+		if(!(prepare_meta_result = mysql_stmt_result_metadata(S->stmt))) {
```

Comment:
I don't think this is a function that should be called on each fetch. Shouldn't the result metadata already be stored inside `S->result`?

Reply:
Makes sense. I replaced it with an existence check on `S->stmt->data->result` which I assume is allowed.